pkg/storage/secret/metadata/outbox_store.go

```diff
@@ -168,16 +189,20 @@ func (s *outboxStore) ReceiveN(ctx context.Context, n uint) ([]contracts.OutboxM
 		messages = append(messages, msg)
 	}
 
-	if err := rows.Err(); err != nil {
```

Comment:
I think we still need the rows.Err() check here after the `rows.Next()`

Reply:
Isn't rows.Scan supposed to return an error?

Reply:
The `rows.Err()` is for cases where there can be errors in the Next() / iteration: https://pkg.go.dev/database/sql#Rows.Next

Reply:
Oh ok, got it, returns false if an error happens. Bad API.

Reply:
The problem would be that `rows.Next()` returns `false` and sets the `lasterror` so you don't call the `Scan` inside the for loop, but you'd have to check it with `rows.Err()`

Reply:
Yep, for some reason rows.Next() doesnt return an `err` itself 😅

Reply:
(code updated to call `rows.Err()`)